### ISSUE TITLE
Errors are leaving pending operations forcing manual intervention

### ISSUE THREAD

**Issue by steinybot** [pulumi/pulumi#5945]
The past couple of days I had several cases where I would be moving things around and creating aliases and something would fail which is fine but then there would be pending operations and if I try and run pulumi again it would tell me that it the CLI was interrupted (which it wasn't) and I would have to manually edit the stack (which I seem to screw up every time and just go on a crazed deletion rampage until it works again).

## Expected Behavior

Errors should not leave pending operations.

## Current Behavior

```
error: the current deployment has 2 resource(s) with pending operations:
  * urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::DatadogFirehoseErrors, interrupted while creating
  * urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:iam/role:Role::DatadogCloudWatch, interrupted while updating

These resources are in an unknown state because the Pulumi CLI was interrupted while
waiting for changes to these resources to complete. You should confirm whether or not the
operations listed completed successfully by checking the state of the appropriate provider.
For example, if you are using AWS, you can confirm using the AWS Console.

Once you have confirmed the status of the interrupted operations, you can repair your stack
using 'pulumi stack export' to export your stack to a file. For each operation that succeeded,
remove that operation from the "pending_operations" section of the file. Once this is complete,
use 'pulumi stack import' to import the repaired stack.

refusing to proceed
```

## Steps to Reproduce

Unfortunately I do not know how to reproduce this. I figured that it has happened enough times now that I should at least raise the issue so that I have somewhere to record what happened.

This most recent time it first failed with:
```
  aws:cloudwatch:LogGroup (DatadogFirehoseErrors):
    error: post-step event returned an error: failed to normalize URN references: Two resources ('urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::DatadogFirehoseErrors' and 'urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::FirehoseErrorLogGroup') aliased to the same: 'urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::FirehoseErrorLogGroup'

  aws:iam:Role (DatadogCloudWatch):
    error: post-step event returned an error: failed to normalize URN references: Two resources ('urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::DatadogFirehoseErrors' and 'urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::FirehoseErrorLogGroup') aliased to the same: 'urn:pulumi:prod::tlayen-infrastructure::tlayen:platform:datadog:Datadog$aws:cloudwatch/logGroup:LogGroup::FirehoseErrorLogGroup'

  pulumi:pulumi:Stack (tlayen-infrastructure-prod):
    error: update failed
    error: Program failed with an unhandled exception:
```

I wonder if the unhandled exception is what is causing it to think that it was interrupted. Perhaps there isn't much that can be done but at the very least it there should be some top level error handler that is able to say that it crashed and not that it was interrupted as that is just very confusing.

The error is weird in and of itself. `FirehoseErrorLogGroup` was renamed to `DatadogFirehoseErrors`. Somehow it is still seeing the old version as an alias of itself. Not sure if this is a separate bug or not.

## Context (Environment)

I'm trying to rename resources.

**Comment by leezen** [comment]
Tracked in https://github.com/pulumi/pulumi/issues/5945

**Comment by lukehoban** [comment]
> Tracked in #5945

@leezen Looks like this was closed as a dupe of itself?  I couldn't find another issue with the same failure mode.

**Comment by leezen** [comment]
@lukehoban Gah, copy and paste fail. Yes. I meant to link to https://github.com/pulumi/pulumi/issues/4265. I'll update the previous comment as well.
